core/src/trace/component.rs

```diff
@@ -0,0 +1,59 @@
+//! Abscissa tracing component
+
+// TODO(tarcieri): logfile support?
+
+use tracing_log::LogTracer;
+use tracing_subscriber::{fmt::Formatter, reload::Handle, EnvFilter, FmtSubscriber};
+
+use super::config::Config;
+use crate::{Component, FrameworkError, FrameworkErrorKind};
+
+/// Abscissa component for initializing the `tracing` subsystem
+#[derive(Component, Debug)]
+#[component(core)]
+pub struct Tracing {
+    filter_handle: Handle<EnvFilter, Formatter>,
+}
+
+impl Tracing {
+    /// Create a new [`Tracing`] component from the given [`Config`].
+    pub fn new(config: Config) -> Result<Self, FrameworkError> {
+        // Configure log/tracing interoperability by setting a `LogTracer` as
+        // the global logger for the log crate, which converts all log events
+        // into tracing events.
+        LogTracer::init().map_err(|e| FrameworkErrorKind::ComponentError.context(e))?;
+
+        // Construct a tracing subscriber with the supplied filter and enable reloading.
+        let builder = FmtSubscriber::builder()
+            // XXX this should use ColorChoice::should_ansi, but where
```

Comment:
Looking at the `log` code that was removed in this PR, it _looks_ like ansi colors were always used (but I may be misreading it?)

Reply:
A lot of the color support regressed when I switched from `term` to `termcolor` (#47). I posted some up-to-date thoughts on it in this comment:

https://github.com/iqlusioninc/abscissa/pull/154#issuecomment-555791977